core/src/main/java/hudson/model/Queue.java

```diff
@@ -195,6 +195,17 @@
     private final ItemList<BlockedItem> blockedProjects = new ItemList<BlockedItem>();
 
     /**
+     * List of type {@link FlyWeightItem}.
+     *
+     * <p>
+     * This consists of {@link Item}s which require one additional executor,
+     * which is a temporary executor, to manage the execution of its nested tasks.
+     * This applies to Matrix project.
+     */
+
+    private final ItemList<FlyWeightItem> flyWeightTasks = new ItemList<FlyWeightItem>();
```

Comment:
@oleg-nenashev I've re-used the buildable list instead in the last pushed implementation, I think it s clearer and lighter now